File: 15-3sum/3sum.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        int i =0;
        int k =nums.size()-1;
        vector<vector<int>> vt ;
        sort(nums.begin(),nums.end());
        while(i<k){
            int j =i+1;
            k=nums.size()-1;
            while(j<k){
                int sum = nums[i]+nums[j]+nums[k];
                if(sum ==0){
                    int x =nums[j];
                    int y =nums[k];
                     vt.push_back({nums[i],nums[j],nums[k]});
                    j++;
                    while(j<k and x==nums[j]){
                        j++;
                    }
                     while(j<k and y==nums[k]){
                       k--;
                    }
                   
                }
                else if (sum>0){
                    k--;
                }
                else {
                    j++;
                }
            }
            int z =nums[i];
            while(i<k and z==nums[i]){
                i++;
            }

        }
            return vt;
    }
};

File: 15-3sum/3sum.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var threeSum = function(nums) {
    const ans=[];
    let i=0;
    let j=i+1;
    nums.sort((a, b) => a - b);
    console.log(nums);
    let k=nums.length-1;
    while(i<k){
        j=i+1;
        k=nums.length-1;
        while(j<k){

            if(nums[j]+nums[i]+nums[k]==0){
            console.log(nums[i]+nums[j]+nums[k]);
                ans.push([nums[i],nums[j],nums[k]]);
                let x=nums[j];
                let y =nums[k];
                while(nums[k]==y){
                    k--;
                }
                 while(nums[j]==x){
                    j++;
                }
            }
             else if (nums[j]+nums[i]+nums[k] > 0){
                
                k--;
            }
            else{
                j++;
            }
        }
          let z =nums[i];
                while(nums[i]==z){
                    i++;
                }
    }
    return ans;
};